src/main/java/com/zilen/weather/service/WeatherService.java

```diff
@@ -25,13 +27,21 @@ public WeatherService(RestTemplate restTemplate, WeatherRepository weatherReposi
         this.weatherConverter = weatherConverter;
     }
 
+    public Weather findByCityNameFromApi(String cityName) {
+        ResponseEntity<Weather> response = restTemplate.getForEntity(url + cityName + "&units=metric&appid=" + appId, Weather.class);
+        weatherRepository.save(weatherConverter.transformToEntity(response.getBody()));
+        return weatherConverter.transformToWeather(weatherRepository.findByName(cityName));
+    }
+
     public Weather findByCityName(String cityName) {
         if (cityName.isBlank()) {
             throw new CityNotFoundException("You must pass the correct cityName!", cityName);
         }
-        ResponseEntity<Weather> response = restTemplate.getForEntity(url + cityName + "&units=metric&appid=" + appId, Weather.class);
-        weatherRepository.save(weatherConverter.transformToEntity(response.getBody()));
-        return weatherConverter.transformToWeather(weatherRepository.findByName(cityName));
+        if (Objects.isNull(weatherRepository.findByName(cityName))) {
```

Comment:
the same situation, you make two `findByName` calls to db, it's not good for performance.